src/java/org/apache/cassandra/utils/vint/VIntCoding.java

```diff
@@ -73,6 +73,21 @@ public byte[] initialValue()
 
     public static final int MAX_SIZE = 10;
 
+    /**
+     * Throw when attempting to decode a vint and the output type
+     * doesn't have enough space to fit the value that was decoded
+     */
+    public static class VIntOutOfRangeException extends RuntimeException
+    {
+        private final long value;
```

Comment:
nit: `value` isn't used?

Reply:
Ah, I meant for that to be public so the caller could inspect the out of range value.

Reply:
Made it public, let me know if you think that makes sense.

Reply:
WFM